core/src/ast/mod.rs

```diff
@@ -0,0 +1,15 @@
+/// As part of the macro expansion and code generation process, Diplomat
+/// generates a simplified version of the Rust AST that captures special
+/// types such as opaque structs, [`Box`], and [`Result`] with utilities
+/// for handling such types.
+mod methods;
+pub use methods::*;
```

Comment:
nit: typical style is to name all items being reexported explicitly (but it's not a strong style preference)